src/session.ts

```diff
@@ -303,6 +305,26 @@ export class SessionManager implements Middleware {
         await this.start();
     }
 
+    /** In Development mode, write the PID to a file where the parent session can find it, to attach the dotnet debugger. */
+    private async writePidIfInDevMode(pwshProcess: PowerShellProcess): Promise<void> {
+        if (this.extensionContext.extensionMode !== vscode.ExtensionMode.Development) { return; }
+        const parentSessionId = process.env.VSCODE_PARENT_SESSION_ID;
+        const pidFilePath = vscode.Uri.joinPath(this.sessionsFolder, `PSES-${parentSessionId}.pid`);
+
+        if (parentSessionId === undefined) { return; }
+
+        const fs = vscode.workspace.fs;
+        const pid = (await pwshProcess.getPid())!.toString();
+        await fs.writeFile(pidFilePath, Buffer.from(pid));
```

Comment:
How annoying lol.

Reply:
I looked at lots of ways to do this, IPC sockets, etc. and this just was the easiest and most effective way.